library/entity/interaction/src/main/java/org/quiltmc/qsl/entity/interaction/api/DamageContext.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2022 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.entity.interaction.api;
+
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.entity.damage.DamageSource;
+import net.minecraft.item.ItemStack;
+
+/**
+ * An encapsulation of several arguments relevant to a damage context with a
+ *     mutable damage value and cancellation that is shared across all listeners.
+ */
+public class DamageContext {
```

Comment:
Annotating members with `@Nullable` and `@NotNull` would be really helpful!

Reply:
Imo, nothing in it should ever be null. Should I just annotate everything with `@notNull`?